### ISSUE TITLE
[Spring Boot]The web Swagger editor can generate successfully but Swagger Codegen CLI couldn't when "allOf" appears under "items" in inline definition

### ISSUE THREAD

**Issue by simingweng** [swagger-api/swagger-codegen#5634]
##### Description
We transformed a IETF Yang model file into Swagger yaml using an open source tool, the resulting YAML can generate a Spring Boot server stub successfully from the online Swagger Editor, but if I run it through swagger-codegen command line, it failed.

##### Swagger-codegen version
2.2.2

##### Swagger declaration file content or url

https://drive.google.com/file/d/0B-x3Vahm51o_Vkpsb0VFQ05aS28/view?usp=sharing

##### Command line used for generation

`java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i ietf-lmap.yaml -l spring -o .`

##### Steps to reproduce

If I upload the YAML in question to http://editor.swagger.io/#/, I can successfully generate and download the "spring" server files

But, if I execute the command line above (I tried with 2.2.2 as well as the master branch), it showed a bunch of warnings like following:

`[main] WARN io.swagger.util.PropertyDeserializer - no property from null, null, {ENUM=null, TITLE=null, DESCRIPTION=The list of tasks configured on the LMAP agent., DEFAULT=null, PATTERN=null, DESCRIMINATOR=null, MIN_ITEMS=null, MAX_ITEMS=null, MIN_PROPERTIES=null, MAX_PROPERTIES=null, MIN_LENGTH=null, MAX_LENGTH=null, MINIMUM=null, MAXIMUM=null, EXCLUSIVE_MINIMUM=null, EXCLUSIVE_MAXIMUM=null, UNIQUE_ITEMS=null, EXAMPLE=null, TYPE=null, FORMAT=null, READ_ONLY=null, VENDOR_EXTENSIONS={}, MULTIPLE_OF=null}`

and errors like following:

```
[main] WARN io.swagger.codegen.DefaultCodegen - null(array property) does not have a proper inner type defined
[main] ERROR io.swagger.codegen.DefaultCodegen - unexpected missing property for name task
[main] WARN io.swagger.codegen.DefaultCodegen - skipping invalid array property {
  "baseName" : "task",
  "getter" : "getTask",
  "setter" : "setTask",
  "name" : "task",
  "defaultValueWithParam" : " = data.task;",
  "baseType" : "List",
  "containerType" : "array",
  "jsonSchema" : "{\n  \"type\" : \"array\"\n}",
  "exclusiveMinimum" : false,
  "exclusiveMaximum" : false,
  "hasMore" : false,
  "required" : false,
  "secondaryParam" : false,
  "hasMoreNonReadOnly" : false,
  "isPrimitiveType" : false,
  "isContainer" : true,
  "isNotContainer" : false,
  "isString" : false,
  "isInteger" : false,
  "isLong" : false,
  "isFloat" : false,
  "isDouble" : false,
  "isByteArray" : false,
  "isBinary" : false,
  "isFile" : false,
  "isBoolean" : false,
  "isDate" : false,
  "isDateTime" : false,
  "isListContainer" : true,
  "isMapContainer" : false,
  "isEnum" : false,
  "isReadOnly" : false,
  "vendorExtensions" : { },
  "hasValidation" : false,
  "isInherited" : false,
  "nameInCamelCase" : "Task"
}
```

It seems the error is thrown when the processor encounter an inline schema definition with an array type whose item type contains a "allOf" syntax.

**Comment by johnnyxh** [comment]
Anyone taken a look at this issue? I have also noticed some differences between files generated from Swagger-Codegen 2.2.2 and the online generator (http://generator.swagger.io/). Both seem to be on version 2.2.2 (assuming the api version reflects the Swagger-Codegen version)

**Comment by wing328** [comment]
@simingweng @johnnyxh the Swagger Editor should be using the latest stable version of Swagger Codegen (v2.2.2)

@fehguy can you help confirm?

**Comment by wing328** [comment]
Please give it another try as we've released v2.2.3 (and Swagger generator has been upgraded to use that)
